lang/python38/Portfile

```diff
@@ -37,7 +37,9 @@ patchfiles          patch-setup.py.diff \
                     patch-Lib-ctypes-macholib-dyld.py.diff \
                     patch-libedit.diff \
                     patch-configure-xcode4bug.diff \
-                    patch-_osx_support.py.diff
+                    patch-_osx_support.py.diff \
+                    patch-no-copyfile-on-Tiger.diff \
+                    patch-threadid-older-systems.diff
```

Comment:
I know its a matter of opinion, but personally I prefer to limit the application of patches to only when they are needed, particularly if the target is only a small number of (older) systems. In this case, as the two patches are only needed on Tiger, I would prefer if the application of them was limited to only these systems, via a `os.major` check.

Reply:
leopard and tiger for the threadid one.

these were written to be always appliable, and should be upstreamable.

applying patches only sometimes is safer but fragile for updates. 

I'm happy either way, whatever Josh wants.

Reply:
Have patches been submitted upstream ? What is their response to them ?

If upstream are receptive to these changes then I would be happier to apply them unconditionally. If not, I would prefer if they remained tiger specific.

Reply:
We haven't submitted these changes upstream yet, because (until this PR) we hadn't fully tested them yet. CPython's support policy for old OS X versions is not very clear (see https://github.com/macports/macports-ports/pull/5987#issuecomment-566798435) so it's difficult to say if they still accept fixes for old systems like 10.4 and 10.5.

Now that they are properly tested (I think we covered every affected OS X version/arch combination), I think it's okay to try submitting them upstream (we'll see if they accept them or not). @kencu If it's okay with you, I can submit both of the patches. (I've made PRs to the CPython repo before, so I already have their contributor agreement signed.)

Reply:
Leopard and Tiger for the threadid one.

Whether the patches are conditional on OS version, and therefore fragile for updates of python38 for the future, I will leave with Josh, as it will be his issue. Previously, MacPorts admins have requested that patches be written so as to be applied unconditionally if possible, as they are more robust that way, and potentially upstreamable.

I know you don't like that policy. It's pretty much up to Josh, as it's his port, and his policy. So I'm good either way. Just don't see a need  to have the discussion every single time with every single port and every patch.